libts/state.go

```diff
@@ -87,10 +109,15 @@ func GetState(ctx context.Context) (State, error) {
 		return State{}, err
 	}
 
+	backendState, err := NewIPNStateFromString(status.BackendState)
+	if err != nil {
+		return State{}, fmt.Errorf("cannot get status from state: %w", err)
+	}
```

Comment:
before, you would hope that some of switch/cases would catch default undetected value (old code did not handle in switches wrong values). but now you error out early on, and enum values are guaranteed to be valid throughout the code.